### PR TITLE
Deprecation Warnings

### PR DESCRIPTION
**Pull request by polytomous**
@staticfloat 

Pretty much everything is boring grepping. The only two areas where I took artistic liberties were defining JULIA_HOME, based on the BINDIR->JULIA_HOME reassignment pending post/during 0.7, and in modifying a global constant for the version and adding some commentary.

### CODE REVIEW

**Comment by polytomous** [comment]
Doesn't look good to merge. :-1: Don't know if real problems or just transient issues. 
Breakdown: 
CI Appveyor: 
Nightly win32: Timed out on build. Looks like there's a million deprecation warnings filling up the log, so it's hard to find if I have a real error there. 

Nightly win64: 
Bug in julia, looks like. Got this message. Looks like it's in the multiple dispatch code. :smile_cat:  
```
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x6b598944 -- jl_typemap_level_assoc_exact at /home/Administrator/buildbot/worker/package_win64/build/src\typemap.c:810
in expression starting at C:\Users\appveyor\.julia\v0.7\Nettle\test\hash_tests.jl:4
jl_typemap_level_assoc_exact at /home/Administrator/buildbot/worker/package_win64/build/src\typemap.c:812
jl_typemap_assoc_exact at /home/Administrator/buildbot/worker/package_win64/build/src\julia_internal.h:944 [inlined]
jl_lookup_generic_ at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2044 [inlined]
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2096
```
CI Travis: 
Nightly ApplE: Looks like #HomeBrewProblems 
```
┌ Error: ------------------------------------------------------------
│ # Build failed for Homebrew
│   exception =
│    LoadError: InitError: failed process: Process(`curl '-#' -L https://github.com/Homebrew/brew/tarball/master`, ProcessExited(6)) [6]
│    Stacktrace:
```

Nightly linux: 
Another bug. 
```
signal (11): Segmentation fault
in expression starting at /home/travis/.julia/v0.7/Nettle/test/hash_tests.jl:4
jl_typemap_level_assoc_exact at /buildworker/worker/package_linux64/build/src/typemap.c:816
jl_typemap_assoc_exact at /buildworker/worker/package_linux64/build/src/julia_internal.h:882 [inlined]
jl_lookup_generic_ at /buildworker/worker/package_linux64/build/src/gf.c:2112 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2164
Type at /home/travis/.julia/v0.7/Nettle/src/hash.jl:23
digest at /home/travis/.julia/v0.7/Nettle/src/hash.jl:44
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:183
```

I have a feeling it's the same as the other type bug found on the other failed, CI. 


Coverage: 
Coverage is down. There already weren't tests exercising V 3 code though, so this just exposes lack of testing, not me causing the lack of testing.

**Comment by staticfloat** [comment]
@polytomous thank you for your work.  :)  I really appreciate help in squashing deprecation warnings and updating code.

I have just merged a PR to get rid of the Homebrew problems (it switches this package over to BinaryProvider rather than Homebrew).  Feel free to rebase your changes on top of this, that should get rid of the need for the `JULIA_HOME` stuff at all.  I've also removed the nettle < 3.x.y code, as since we're using BinaryProvider now, we are guaranteeing ourselves that our `libnettle` is much more recent (3.4 now)

**Comment by polytomous** [comment]
Sorry for the delay, I had a few things pop up. 

I've got your changes integrated, but I ran into a new issue as I was running tests so I'm stepping through that at the moment. 

On the latest julia version, I'm getting a segfault for calls to the hashtype init function in libnettle, i'm a bit out of my depth here, so i'm hoping this is a transient issue. I think it's likely I just have a dev environment issue or it's something wrong i introduced in the build.jl file. My error in case you have any insight:

```
signal (11): Segmentation fault
in expression starting at /home/theo/github/polytomous/Nettle/test/hash_tests.jl:4
jl_typemap_level_assoc_exact at /home/theo/github/JuliaLang/julia/src/typemap.c:816
jl_typemap_assoc_exact at /home/theo/github/JuliaLang/julia/src/julia_internal.h:882 [inlined]
jl_lookup_generic_ at /home/theo/github/JuliaLang/julia/src/gf.c:2112 [inlined]
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2164
Type at /home/theo/github/polytomous/Nettle/src/hash.jl:28
digest at /home/theo/github/polytomous/Nettle/src/hash.jl:49
jl_fptr_trampoline at /home/theo/github/JuliaLang/julia/src/gf.c:1838
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2167
top-level scope at ./<missing>:40
jl_fptr_trampoline at /home/theo/github/JuliaLang/julia/src/gf.c:1838
jl_toplevel_eval_flex at /home/theo/github/JuliaLang/julia/src/toplevel.c:850
jl_parse_eval_all at /home/theo/github/JuliaLang/julia/src/ast.c:852
jl_load at /home/theo/github/JuliaLang/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2167
include at ./sysimg.jl:68
jl_fptr_trampoline at /home/theo/github/JuliaLang/julia/src/gf.c:1838
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2167
do_call at /home/theo/github/JuliaLang/julia/src/interpreter.c:324
eval_value at /home/theo/github/JuliaLang/julia/src/interpreter.c:416
eval_stmt_value at /home/theo/github/JuliaLang/julia/src/interpreter.c:367 [inlined]
eval_body at /home/theo/github/JuliaLang/julia/src/interpreter.c:700
jl_interpret_toplevel_thunk_callback at /home/theo/github/JuliaLang/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7fbcfd0e5f2f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /home/theo/github/JuliaLang/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/theo/github/JuliaLang/julia/src/toplevel.c:856
jl_parse_eval_all at /home/theo/github/JuliaLang/julia/src/ast.c:852
jl_load at /home/theo/github/JuliaLang/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2167
exec_options at ./client.jl:327
_start at ./client.jl:455
jl_apply_generic at /home/theo/github/JuliaLang/julia/src/gf.c:2167
jl_apply at /home/theo/github/JuliaLang/julia/ui/../src/julia.h:1540 [inlined]
true_main at /home/theo/github/JuliaLang/julia/ui/repl.c:109
main at /home/theo/github/JuliaLang/julia/ui/repl.c:240
__libc_start_main at /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
_start at /home/theo/github/JuliaLang/julia/usr/bin/julia (unknown line)
Allocations: 4237019 (Pool: 4235420; Big: 1599); GC: 9
Segmentation fault (core dumped)
```

I'm 100% sure that my latest changes will not pass CI, but I'm pushing them any way and kind of hoping someone will have some tips for debugging libnettle issues.

**Comment by polytomous** [comment]
I will say at a minimum that you can probably cherry pick over my `manifest.toml` and `project.toml` updates. `Project.toml` also needs a better description. Realistically you should pick what you want to put there, I just picked something on a whim from the Readme.

File: src/Nettle.jl [hunk 1]
```diff
@@ -21,6 +21,7 @@ function __init__()
 end
 
 # SnoopCompile acceleration
+using Libdl
```
Comment (staticfloat) [inline src/Nettle.jl]: This needs some `Compat` magic to work on 0.6.

**Comment by staticfloat** [comment]
> On the latest julia version, I'm getting a segfault for calls to the hashtype init function in libnettle

Yep, this is a known issue unfortunately, reported here: https://github.com/JuliaLang/julia/issues/26892.  As far as I can tell, this is a bug in `julia`.  Let's ignore it for now, and focus on getting 0.6 working with as few depwarns as possible from 0.7.

**Comment by polytomous** [comment]
So... 
I cut the nettle.jl dependency in the library that was calling it and moved on. 😅😅😅😅 

Feel free to use this code if you want, but I'm just going to close the PR; sorry for the extra noise.

**Pull request closed without merging**